tests/analytical_comparisons/test_analytical.py

```diff
@@ -88,9 +105,26 @@ def test_analytical(name, expected, config):
     convergence = np.log2(errs.shift() / errs).drop(index=0)
     expected_convergence = pd.Series(expected["convergence"], index=cols_err)
 
-    # Make sure velocity and pressure have the theoretical convergence rates
-    assert np.allclose(convergence[columns[:2]], expected_convergence[columns[:2]], rtol=expected.get("rtol", 1e-2))
-
-    # For normal stress/free surface we only make sure the convergence does not go bellow a minimum, while allowing loose tolerance
-    assert np.allclose(convergence[columns[2]], expected_convergence[columns[2]], rtol=expected.get("rtol_ns", 1e-1))
-    assert np.all(convergence[columns[2]] > expected_convergence[columns[2]] - 0.1)
+    if name.split("/")[-1] == "free_surface":
+        # Make sure velocity, pressure, and free surface have the theoretical rates
+        assert np.allclose(
+            convergence, expected_convergence, rtol=expected.get("rtol", 1e-2)
+        )
+    else:
+        # Make sure velocity and pressure have the theoretical rates
+        assert np.allclose(
+            convergence[cols_err[:-1]],
+            expected_convergence[cols_err[:-1]],
+            rtol=expected.get("rtol", 1e-2),
+        )
+        # Make sure normal stress convergence is bounded
+        assert np.all(
+            (
+                convergence[cols_err[-1]]
+                > expected_convergence.iloc[-1] - expected.get("ns_lb", 1e-2)
+            )
+            & (
+                convergence[cols_err[-1]]
+                < expected_convergence.iloc[-1] + expected.get("ns_ub", 1e-2)
+            )
+        )
```

Comment:
Can this be two `assert` statements? It would make it a little easier to see where a failure arises.